### PR TITLE
Add recurse keyword to Collection.iter_contents

### PR DESCRIPTION
**Pull request by kjsanger**
Sort the result of Collection.iter_contents.
Sort some arrays in place to avoid copying.
Improve some variable names by avoiding abbreviations.

### CODE REVIEW

**Comment by mksanger** [comment]
To make it clear, this is a breaking change, since the default behaviour changes from recursive to non-recursive.

**Comment by kjsanger** [comment]
The previous behaviour of contents `iter_contents` was to show the immediate contents of a collection (so, like `contents`, but as an iterator). The new behaviour adds a `recurse` flag that defaults to false, so I don't think that breaks old code.

**Comment by mksanger** [comment]
That is not correct, if you look at the diff, the behaviour that is inside the recurse flag conditional is the same as the behaviour that was previously present

File: src/partisan/irods.py [hunk 1]
```diff
@@ -1976,14 +1978,25 @@ def iter_contents(
...
+                        acl=acl,
+                        avu=avu,
+                        recurse=recurse,
+                        timeout=timeout,
+                        tries=tries,
+                    )
+        else:
+            for elt in contents:
+                elt.path = self.path / elt.path
+                yield elt
```
Comment (mksanger) [inline src/partisan/irods.py:1999]:
```suggestion

        for elt in contents:
            elt.path = self.path / elt.path  # Make an absolute path
            yield elt
            if recurse:
                if isinstance(elt, Collection):
                    yield from elt.iter_contents(
                        acl=acl,
                        avu=avu,
                        recurse=recurse,
                        timeout=timeout,
                        tries=tries,
                    )
```
I also suggest that the conditional be inside the for loop, to remove repeated code.

Comment (kjsanger) [reply src/partisan/irods.py:1999]: Oh, yes. You're right - that was actually a bug, then! Originally I intended it to be non-recursive, but I think I ended up putting some experimental recursive code in there earlier than I wanted to.

**Comment by kjsanger** [comment]
I was also not paying attention to the test that showed it was recursing, originally ...

**Pull request merged**